mobie/metadata/slice_grid_view.py

```diff
@@ -0,0 +1,93 @@
+import warnings
+from elf.io import open_file
+
+from . import view_metadata as view_utils
+from .dataset_metadata import read_dataset_metadata, write_dataset_metadata
+
+
+# TODO handle online sources and refactor this to somewhere more generic
+# generally very useful!!!
+def _load_shape(image_data):
+    pass
+
+
+def _load_transformation():
+    pass
+
+
+# load the scale for the image data. if axis is given only for that axis.
+# use _load_transformation to get the full transformation from the iamge metadata
+# TODO handle online sources and refactor this to somewhere more generic
+# generally very useful!!!
+def _load_scale(image_data, axis=None):
+    pass
+
+
+def create_slice_grid(dataset_folder, source, n_slices, view_name, axis=0, overwrite=False):
+    """Create a grid that aligns n slices of a source shifted so that each n-th slice is aligned with the origin.
+
+    Arguments:
+        dataset_folder [str] -
+        source [str] -
+        n_slices [int] -
+        view_name [str] -
+        axis [int] -
+        overwrite [bool] - whether to overwrite existing views (default: False)
+    """
+    if axis != 0:
+        raise NotImplementedError("create_slice_grid is currently only supported along the z-axis (axis 0)")
+
+    ds_meta = read_dataset_metadata(dataset_folder)
+    sources = ds_meta["sources"]
+    views = ds_meta["views"]
+
+    if source not in sources:
+        raise ValueError(f"The source {source} could not be found in the sources of the dataset at {dataset_folder}.")
+    if view_name in views:
+        msg = "The view {name} is alread present in the dataset at {dataset_folder}."
+        if overwrite:
+            warnings.warn(msg + " It will be over-written.")
+        else:
+            raise ValueError(msg)
+
+    source_type, source_data = next(iter(sources[source].items()))
+    if source_type != "imageSource":
+        raise ValueError("create_slice_grid is only supported for image sources.")
+
+    shape = _load_shape(source_data["imageData"])
```

Comment:
Is this intended to happen in original voxel space or in physical space?
My idea was to slice along physical axes (making it compatible also with any kind of transformed sources) and simply introduce translations by `source.shape[axis]/ n_slices`. The case of a source only being scaled to physical space and slicing happening in a source axis would then be a subset of that general slicing operation.

Reply:
:) No worries

Reply:
So we will need to apply all existing transformations to the "shape" vector. The slicing step would then be `np.dot(T_all,shape)[axis]/n`.

Reply:
I see. That's more elegant and versatile than what I am doing right now. In the current implementation draft you can only apply this function to a single source though, which usually just contains the scale. Should I add another parameter to the function so that you can pass a list of transformations that's applied *before* the slicing? That would make it even more versatile. (I can't support crop in the beginning though, but I would check for that and throw a non-implemented error; and grid won't make sense in general and produce a value error)

Reply:
I think this is what a user would expect and it would enable slicing of composited registered multi-modal sources. So you could display the full platy content sliced evenly-spaced in any direction.

Reply:
I think applying the slicing function to a single ("mother") source would be perfectly fine. You could simply add the other sources to the grid view and its transforms later once they are generated.

Reply:
> I think applying the slicing function to a single ("mother") source would be perfectly fine. You could simply add the other sources to the grid view and its transforms later once they are generated.

This makes perfect sense, but I am not sure we need the axis parameter then. Wouldn't you always slice along the z axis? (w.r.t. the initial transformation of course)

Reply:
Slicing across another axis would just mean rotate it s.t. this axis becomes z

Reply:
> Should I add another parameter to the function so that you can pass a list of transformations that's applied _before_ the slicing?

Maybe you could even reference an existing view and "import" the sourceTransforms from there?

Reply:
> This makes perfect sense, but I am not sure we need the axis parameter then. Wouldn't you always slice along the z axis? (w.r.t. the initial transformation of course)

Yes, you would then add the extra rotation in case slicing another axis. Otherwise the grid will also not be in the `xy` plane. Probably referencing an existing view (that users can generate in MoBIE and export) would be indeed ideal here (and then we can ditch the `axis` parameter). The users probably don't care how the transforms look as long as the orientation matches with what they like to see.